### DISCUSSION TITLE
Login screen shows popup even without using SSO while session expired!

### DISCUSSION THREAD

**Discussion by tarikrital** [umbraco/Umbraco-CMS#17522] [Features and ideas]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

14.2.0

### Bug summary

![image](https://github.com/user-attachments/assets/64f0f713-b575-4636-83d6-73b23e39be5b)
I see the screen shot + title is enough.

### Steps to reproduce

After keeping the screen idle 30+ mins, trying to login a popup shows.

Why having a popup, this is unusual? (low UX)

**Comment by enkelmedia** [comment]
This happens to me when I run different Umbraco instances on localhost, effectively they would overwrite each others cookies. Could that be the problem that you are facing?

**Reply by tarikrital** [reply]
This ticket is about the UX! i.e while logging using V14 it shows second screen of the old Umbraco version then going to home page!


https://github.com/user-attachments/assets/095ffbac-b177-4552-a04f-83b5b8a3092d

**Comment by iOvergaard** [comment]
@wpplumber From what I can gather, you are addressing two concerns in this discussion:

1) There is a popup when you timeout
2) You see a flash of an older background after login

The first concern (the popup) happens because the login screen is a popover lying over your content. After a successful login, the popover will disappear preserving whatever data you were editing. I agree with you that this is not the nicest UX, even though it works. An idea could be to integrate the login form itself as an iframe. As you know, the login form could be either Umbraco or any external provider like Umbraco Id or Google, Azure, etc.

The second concern (flash of an older background) has been addressed for 15.1 with #17453. The first Release Candidate is out now and the final version will follow in two weeks.

Do you have any remarks or ideas for how the UX could be improved regarding the popup?

**Reply by tarikrital** [reply]
My idea here is to show popup only for external providers which seems correct as something not developed by Umbraco, beside that I don't know why you mentioned the `iframe` solution! is it an obligation?

**Reply by iOvergaard** [reply]
Yes, we cannot just give you the login screen because of the OpenID flow; If you were to go directly to `/umbraco/login` you would not be able to log in because your authorization code would not have been generated yet. You must first go to the client requesting the login (backoffice, swagger-ui, postman, etc.) and then be sent to the `/auth` endpoint of the Management API, which figures out if you already have a cookie or need to be sent to the login screen. In this regard, Umbraco acts like an external login provider, which also allows you to log in from a NodeJS app, Postman, or your own client, etc.

**Reply by tarikrital** [reply]
Correct 👍 

Yes, I use it while testing Umbraco package with Swagger.
![image](https://github.com/user-attachments/assets/c91154b3-e68f-41b2-ad39-30d0e687ee02)

Then the iframe is the excellent solution.

Thanks for sharing the info above.
